README.md

```diff
@@ -162,6 +162,17 @@ the `encryption_*` and `ssl_*` options
 directly to the constructor as keyword arguments, even though it is
 required when they are placed in the environment file.
 
+Creating PAM or Native Credentials File (.irodsA)
+-------------------------------------------------
+
+```Two free functions exist for creating encoded authentication files.
+    irods.client_init.write_credentials_with_native_password
+    irods.client_init.write_credentials_with_pam_password
+```
+
+Each takes a cleartext password and creates an appropriately encoded version of it
+as an .irodsA file in the login environment, if none is pre-existing.
```

Comment:
What happens if the .irodsA file exists?

Reply:
> What happens if the .irodsA file exists?

It will definitely overwrite it.  So I should probably change the wording here....

Reply:
revised.

Reply:
I need to revisit this and ask the right questions, it seems.  In the `write_pam_credentials_to_secrets_file` function, I currently assert that the `auth_file` attribute of the session is null (which means the .irodsA file didn't exist or else was not used to authenticate the session.)

 What I probably need to be doing is the following: check that the computed location of the .auth file (whether or not it is pre-existing) is not already there before writing to it.  And the computation of that path probably ought to be based on, in this order
   - what auth file path if any was used to authenticate the session
   - what the IRODS_AUTHENTICATION_FILE environment variable dictates to be the path
   - the path of a hypothetical .irodsA that would be co-located with the environment file used to configure the session
Would appreciate if @trel @alanking and @korydraughn could corroborate this potential approach.

Reply:
Hmm, writing the auth file is independent of any session so I think the first bullet item isn't something you need to worry about. You're either writing to the default location or `$IRODS_AUTHENTICATION_FILE` is defined.

Reply:
Maybe so.   Here, I'm just trying to ensure the logic of the choice of what absolute path we choose to write into ( it's usually ~/.irods/.irodsA,  but  could vary) is consistent between its implementation around the session creation and other usages, like creating .irodsA via the free function.  Fortunately this is trivial, because the free function in this case actually needs to create the session object and communicate with the server anyway, in order to calculate the credentials string to be written.

Reply:
Done and tested.